### ISSUE TITLE
obs_scene_create doesn't make names unique despite what's written in the documentation

### ISSUE THREAD

**Issue by adriansuwala** [obsproject/obs-studio#4076]
## Platform

Operating system and version: Windows 10: Build 19042, release: 2009, revision: 685, 64bit
OBS Studio version: 26.1.1

## Expected Behavior

Name of the second scene is made unique as described in the documentation (https://obsproject.com/docs/reference-scenes.html#c.obs_scene_create)

## Current Behavior

Name is not changed, there are multiple scenes with the same name.

## Steps to Reproduce

1. Save this code as a Lua script and load it
```lua
obs = obslua

function script_properties()
    properties = obs.obs_properties_create()
    obs.obs_properties_add_button(properties, "test", "Test", test)
    return properties
end

function test(properties, property)
    local t = obs.obs_scene_create("test")
    obs.obs_scene_release(t)
    t = obs.obs_scene_create("test")
    obs.obs_scene_release(t)
end
```
2. Click the Test button

**Comment by RytoEX** [comment]
Confirmed that the script provided exhibits the "Current Behavior" described in OBS Studio 28.0.3 on Windows 10.

**Comment by derrod** [comment]
This should be fixed now with #8229, which forces public source names to be unique due them being part of a hash table.
